### PR TITLE
Feat: Explorer Social Follows v1

### PR DESCRIPTION
**Pull request by kaushalmeena**

### CODE REVIEW

File: src/Components/AddressesModal.tsx [hunk 1]
```diff
@@ -2,47 +2,33 @@ import { Modal } from './Modal';
...
-            <div
-              className="mb-8 px-3 py-1 rounded-18 ellipsis hover:bg-glass cursor-pointer"
-              key={index}
-              onClick={() => onAddressClick(value)}
-            >
-              {value}
-            </div>
-          ))}
-        </div>
+      <div className="w-[600px] max-h-[60vh] h-auto bg-primary rounded-xl p-5 overflow-auto grid grid-cols-2 gap-y-5">
```
[the remark this answers is not in the document]
Comment (kaushalmeena) [reply src/Components/AddressesModal.tsx:22]: I am using grid to show 2 column view here, the advantage here is that we don't have to create columns with this approach manually

File: src/Components/Dropdown.tsx [hunk 2]
```diff
@@ -13,12 +14,17 @@ export function Dropdown({
...
+  heading
 }: {
   options: Option[];
   selected?: Option[];
   closeOnSelect?: boolean;
-  renderPlaceholder: (option: Option[], isOpen: boolean) => ReactNode;
+  renderPlaceholder: (
+    option: Option[],
+    isOpen: boolean,
+    isDisabled?: boolean
```
[the remark this answers is not in the document]
Comment (kaushalmeena) [reply src/Components/Dropdown.tsx:26]: renderPlaceholder can render any custom component, there was a need to add disabled styles to that component

File: src/Components/GetAPIDropdown.tsx [hunk 3]
```diff
@@ -1,76 +1,101 @@
...
-              key={label}
-            >
-              {label}
-            </a>
-          ))}
+          <Icon name="tools" className="mr-1" height={16} width={16} />
+          Get API
+        </button>
+        {isDropdownVisible && (
+          <div
```
[the remark this answers is not in the document]
Comment (kaushalmeena) [reply src/Components/GetAPIDropdown.tsx:65]: actually I need conditional styles on dropdown itself
```
<div
            className={classNames(
              'bg-glass rounded-18 p-1 mt-1 flex flex-col absolute min-w-[214px] top-9 z-20',
              {
                'left-0': dropdownAlignment === 'left',
                'left-1/2 -translate-x-1/2': dropdownAlignment === 'center',
                'right-0': dropdownAlignment === 'right'
              }
            )}
            onClick={handleDropdownClose}
          >
```

File: src/Components/Filters/BlockchainFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
+/* eslint-disable react-refresh/only-export-components */
+import { useCallback, useEffect, useMemo } from 'react';
+import { useSearchInput } from '../../hooks/useSearchInput';
+import { Dropdown, Option } from '../Dropdown';
+import { FilterOption } from './FilterOption';
+import { FilterPlaceholder } from './FilterPlaceholder';
+
+export const enum BlockchainFilterType {
```
[the remark this answers is not in the document]
Comment (kaushalmeena) [reply src/Components/Filters/BlockchainFilter.tsx:8]: have fixed it in next commit

File: src/Components/Filters/SortBy.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable react-refresh/only-export-components */
+import { useCallback, useEffect, useMemo } from 'react';
+import { useSearchInput } from '../../hooks/useSearchInput';
+import { Dropdown, Option } from '../Dropdown';
+import { FilterOption } from './FilterOption';
+import { FilterPlaceholder } from './FilterPlaceholder';
+
+export const enum SortOrderType {
```
[the remark this answers is not in the document]
Comment (kaushalmeena) [reply src/Components/Filters/SortBy.tsx:8]: have fixed it in next commit

File: src/pages/TokenBalances/SocialFollows/TableRow.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  onAddressClick: (address: string, dataType?: string) => void;
+}) {
+  const _wallet = item.followerAddress || item.followingAddress;
+
+  const _tokenId = item.followerProfileId || item.followingProfileId;
+
+  const getShowMoreHandler = (values: string[], type: string) => () =>
+    onShowMoreClick(values, type);
+
+  const _social = _wallet?.socials?.find(
```
[the remark this answers is not in the document]
Comment (kaushalmeena) [reply src/pages/TokenBalances/SocialFollows/TableRow.tsx:35]: have fixed it next commit

File: src/pages/TokenHolders/index.tsx [hunk 7]
```diff
@@ -92,51 +90,53 @@ export function TokenHolders() {
...
     const requestFilters = getRequestFilters(tokenFilters);
+    const _hasSocialFilters = Boolean(requestFilters?.socialFilters);
+    const _hasPrimaryDomain = requestFilters?.hasPrimaryDomain;
+    if (address.length === 0) return '';
     if (address.length === 1) {
       return getNftOwnersQueryWithFilters(
         address[0].address,
-        Boolean(requestFilters?.socialFilters),
-        requestFilters?.hasPrimaryDomain
+        _hasSocialFilters,
```
[the remark this answers is not in the document]
Comment (kaushalmeena) [reply src/pages/TokenHolders/index.tsx:120]: have fixed it next commit

File: src/queries/socialFollowersQuery.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+}: {
+  queryFilters: SocialFollowQueryFilters;
+  logicalFilters: SocialFollowLogicalFilters;
+}) => {
+  const _variables = [
+    '$identity: Identity!',
+    '$dappName: SocialFollowDappName',
+    '$limit: Int'
+  ];
+  const _socialFilters = [];
```
[the remark this answers is not in the document]
Comment (kaushalmeena) [reply src/queries/socialFollowersQuery.ts:18]: have fixed it next commit

File: src/queries/socialFollowingQuery.ts [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+}: {
+  queryFilters: SocialFollowQueryFilters;
+  logicalFilters: SocialFollowLogicalFilters;
+}) => {
+  const _variables = [
+    '$identity: Identity!',
+    '$dappName: SocialFollowDappName',
+    '$limit: Int'
+  ];
+  const _socialFilters = [];
```
[the remark this answers is not in the document]
Comment (kaushalmeena) [reply src/queries/socialFollowingQuery.ts:18]: have fixed it next commit

File: src/Components/AddressesModal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aadilhasan) [reply src/Components/AddressesModal.tsx:22]: nice i like that

**Review by aadilhasan — APPROVED** [review]

**Pull request merged**